types/parameterized_single_integer_param_type.go

```diff
@@ -0,0 +1,101 @@
+package types
+
+import (
+	"fmt"
+
+	"github.com/substrait-io/substrait-go/types/parameter_types"
+)
+
+// parameterizedTypeSingleIntegerParam This is a generic type to represent parameterized type with a single integer parameter
+type parameterizedTypeSingleIntegerParam[T VarCharType | FixedCharType | FixedBinaryType | PrecisionTimestampType | PrecisionTimestampTzType] struct {
```

Comment:
we should shift these to a constraint to shorten this line and make it easier for reuse. It also would allow us to add the common functions to the interface constraint and make the subsequent code simpler.

```go
type singleIntParamType interface {
        VarCharType | FixedCharType | FixedBinaryType |
                PrecisionTimestampType | PrecisionTimestampTzType

         ShortString() string
         BaseString() string
}

type parameterizedTypeSingleIntegerParam[T singleIntParamType] struct {
```

Reply:
fixed it